test/models/applicationCost.model.test.js

```diff
@@ -168,6 +168,11 @@ lab.experiment('Application Cost Model test:', () => {
       Code.expect(applicationCost.items.length).to.equal(8)
       Code.expect(spy.callCount).to.equal(2)
     })
+    lab.test('getApplicationCostForApplicationId clears cached application data', async () => {
+      context.application = fakeApplicationEntity
```

Comment:
I would maybe have a test for "retrieves correct current cost value" (the caching was the cause of the error, not the error itself).
So on this line instead of using `fakeApplicationEntity`, create one with an incorrect or blank cost value and put that in the context, then check that the call to `getApplicationCostForApplicationId` has returned an object with the correct `TOTAL_COST` value.

Something like:
```
    context.application = new ApplicationEntity({
      id: FAKE_APPLICATION_ID,
      applicantType: DYNAMICS_PERMIT_HOLDER_TYPES.PARTNERSHIP.dynamicsApplicantTypeId,
      organisationType: DYNAMICS_PERMIT_HOLDER_TYPES.PARTNERSHIP.dynamicsOrganisationTypeId,
      lineItemsTotalAmount: TOTAL_COST + 1
    })
```
and then after the call:
`Code.expect(applicationCost.total.cost).to.equal(TOTAL_COST)`

Reply:
Thanks Ben, I've revised the test to be more along these lines.